src/sweeps/bayes_search.py

```diff
@@ -302,7 +299,18 @@ def next_sample(
         best_test_X_index = np.argmax(prob_of_improve)
     else:
     """
-    best_test_X_index = np.argmax(e_i)
+
+    # Check to see that the metric varies accross the samples
+    if np.all(np.isclose(filtered_y, filtered_y[0], atol=METRIC_VALUE_ATOL)):
+        logging.warning(
+            f"All instances of metric are within the minimum tolerance of {METRIC_VALUE_ATOL},"
+            + "the next sample will be a random sample within parameter space"
+        )
+        best_test_X_index = np.random.randint(0, test_X.shape[0] - 1)
+    else:
+        # Round for numerical stability
+        e_i = np.around(e_i, decimals=EXPECTED_IMPROVEMENT_SIGNIFICANT_DECIMAL_PLACES)
```

Comment:
with the above we should no longer need this. if we have to use this, we are relying on numerical instabilities 

```suggestion
```